File: hubbo-parent/hubbo-dal/src/main/java/cn/hubbo/dal/config/DatasourceConfiguration.java
package cn.hubbo.dal.config;

import com.alibaba.druid.pool.DruidDataSource;
import org.apache.ibatis.session.ExecutorType;
import org.apache.ibatis.session.SqlSessionFactory;
import org.mybatis.spring.SqlSessionFactoryBean;
import org.mybatis.spring.SqlSessionTemplate;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.env.Environment;

import javax.sql.DataSource;

@Configuration
public class DatasourceConfiguration {

	@Autowired
	private Environment environment;

	@Bean
	public DataSource dataSource() {
		DruidDataSource dataSource = new DruidDataSource();
		dataSource.setAsyncInit(true);
		dataSource.setUsername(environment.getProperty("spring.datasource.username"));
		dataSource.setPassword(environment.getProperty("spring.datasource.password"));
		dataSource.setUrl(environment.getProperty("spring.datasource.url"));
		dataSource.setDriverClassName(environment.getProperty("spring.datasource.driver-class-name"));
		return dataSource;
	}

	@Bean
	public SqlSessionFactory sqlSessionFactory() throws Exception {
		SqlSessionFactoryBean sqlSessionFactoryBean = new SqlSessionFactoryBean();
		sqlSessionFactoryBean.setDataSource(dataSource());
		return sqlSessionFactoryBean.getObject();
	}


}

File: hubbo-parent/hubbo-configure/src/main/kotlin/cn/hubbo/config/db/DatasourceConfiguration.kt
package cn.hubbo.config.db

import io.r2dbc.pool.ConnectionPool
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Lazy
import org.springframework.context.annotation.Profile


//  todo r2dbc连接池配置
@Configuration
class DatasourceConfiguration {

    @Profile("dev")
    @Lazy
    @Bean("connectionPool")
    fun devConnectionPool(): ConnectionPool? {
        return null
    }


    @Profile("!dev")
    @Bean("connectionPool")
    fun connectionPool(): ConnectionPool? {
        return null
    }


}